tools/cloud_functions/gcs_event_based_ingest/backfill.py

```diff
@@ -0,0 +1,162 @@
+                future_to_gsurl[executor.submit(
+                    ps_cli.publish,
+                    args.pubsub_topic,
+                    b'',    # cloud function ignores message body
+                    bucketId=blob.bucket.name,
+                    objectId=blob.name,
+                    _metaInfo="this message was submitted with "
+                    "gcs_ocn_bq_ingest backfill.py utility"
+                )] = f"gs://{blob.bucket.name}/{blob.name}"
+            else:
+                logging.info("running  cloud function locally for: %s",
+                             f"gs://{blob.bucket.name}/{blob.name}")
+                future_to_gsurl[executor.submit(
+                    gcs_ocn_bq_ingest.main.main,
+                    {
+                        "attributes": {
+                            "bucketId": blob.bucket.name,
+                            "objectId": blob.name
+                        }
+                    },
+                    None,
+                )] = f"gs://{blob.bucket.name}/{blob.name}"
+        exceptions: Dict[str, Exception] = dict()
+        for future in concurrent.futures.as_completed(future_to_gsurl):
+            gsurl = future_to_gsurl[future]
+            try:
+                future.result()
+            except Exception as err:  # pylint: disable=broad-except
+                logging.error("Error processing %s: %s", gsurl, err)
+                exceptions[gsurl] = err
+        if exceptions:
+            raise RuntimeError("The following errors were encountered:\n" +
+                               pprint.pformat(exceptions))
+
+
+def parse_args(args: List[str]) -> Namespace:
+    """argument parser for backfill CLI"""
+    parser = ArgumentParser(
+        description="utility to backfill success file notifications "
+                    "or run the cloud function locally in concurrent threads.")
+
+    parser.add_argument(
+        "--gcs_path",
+        "-p",
+        help="GCS path (e.g. gs://bucket/prefix/to/search/)to search for "
+        "existing _SUCCESS files",
+        required=True,
+    )
+
+    parser.add_argument(
+        "--mode",
+        "-m",
+        help="How to perform the backfill: LOCAL run cloud function main"
+        " method locally (in concurrent threads) or NOTIFICATIONS just push"
+        " notifications to Pub/Sub for a deployed version of the cloud function"
+        " to pick up. Default is NOTIFICATIONS.",
+        required=False,
+        type=str.upper,
+        choices=["LOCAL", "NOTIFICATIONS"],
+        default="NOTIFICATIONS",
+    )
+
+    parser.add_argument(
+        "--pubsub_topic",
+        "--topic",
+        "-t",
+        help="Pub/Sub notifications topic to post notifications for. "
+        "i.e. projects/{PROJECT_ID}/topics/{TOPIC_ID} "
+        "Required if using NOTIFICATIONS mode.",
+        required=False,
+        default=None,
+    )
+
+    parser.add_argument(
+        "--success_filename",
```

Comment:
```suggestion
        "--success-filename",
```
This is the common convention.